### ISSUE TITLE
Is there any way to morph the images without cropping just the face?

### ISSUE THREAD

**Issue by OtavioRMachado** [alyssaq/face_morpher#3]

**Comment by alyssaq** [comment]
Do you mean morphing something other than a face? 
Opencv's face detector does require the face to be fairly prominent in the image but you shouldn't need to crop the face.

**Comment by OtavioRMachado** [comment]
No, morphing faces, but with the rest of the image too. 
I'd like to reposition the faces, but don't want to delete the rest of the image around the faces.

**Comment by alyssaq** [comment]
Ah, I see. It is possible and can be complex depending on the images. My version currently only uses the face points detected from [stasm](http://www.milbo.users.sonic.net/stasm/). It uses openCV's eye, mouth and face detector. 

To morph the rest of the image, you would have to add more points surrounding the outside of the face. Getting it consistent across all face images will be tricky. It would require detecting and aligning the neck, collar, hair, etc. and handle all inconsistencies of the background across images.

Alternatively, you can manually define your own control points for all images and supply the array of points into locator.py.

Alyssa

**Comment by b3qkc** [comment]
Hi, Alyssa.
First of all I would like to thank you for this great app! No hand driven triangulation, resizing, etc. But maybe it's possible to crossfade the rest of the image as background with no geometry changes? We mostly looking at the face while magic is going and having some background on video is better than black nothing. Unfortunately I have not enough programmer skills to modify the program that way but I think it would be useful improvment!

Slava

**Comment by alyssaq** [comment]
Hi Slava

Thanks for the comment. Do you mean adding a static background image instead of the black?

**Comment by b3qkc** [comment]
Not quite.. I mean crossfade entire images as background and place face morphing over it. I've made some examples..
![background](https://cloud.githubusercontent.com/assets/13962180/9486935/c5a6032e-4bd5-11e5-9cfd-a02682b2eb5c.gif)
![face_alpha](https://cloud.githubusercontent.com/assets/13962180/9486938/cad11fc8-4bd5-11e5-939f-982ad98773a5.gif)
![face_over_backgnd](https://cloud.githubusercontent.com/assets/13962180/9486940/ce7fe424-4bd5-11e5-85a9-73aa6a05ec5a.gif)

**Comment by alyssaq** [comment]
What you've done seems to be alright. Its the reason the app provides an alpha background --- for users to put any background. 

I like your use case but probably not going to implement this cross-fade effect as its user/application specific. Its also more of a photoshop effect rather than the scientific implementation of morphing (which was the goal of this project). That said, I really like the last gif!

**Comment by b3qkc** [comment]
Please, please, please!...

**Comment by azazel75** [comment]
I've done just that in my branch... ;-)
